cnd/src/network/mod.rs

```diff
@@ -136,6 +138,81 @@ impl Swarm {
             local_peer_id,
         })
     }
+
+    /// This is the API for Alice to call in order to execute the appropriate
+    /// communication protocols to announce a swap to Bob (i.e., to send the
+    /// known swap parameters and receive the remaining swap parameters) in
+    /// order to finalize this swap.
+    ///
+    /// This works for all supported combinations on Han, HALight, and Herc20.
+    pub async fn finalize_announce_swap<A, B>(
+        &mut self,
+        _digest: SwapDigest,
+        _announce: AnnounceSwapParameters<A, B>,
+    ) -> anyhow::Result<AnnouncedSwapParameters<A, B>> {
+        // let mut _swarm = self.swarm.lock().await;
+        // swarm.behaviour.finalize_announce_swap(_digest, _announce).await?
+
+        unimplemented!()
+    }
+
+    /// This is the API for Bob to call in order to execute the appropriate
+    /// communication protocols when an announced swap is received from Alice
+    /// (i.e., to receive Alice's known swap parameters and send the remaining
+    /// swap parameters) in order to finalize this swap.
+    ///
+    /// This works for all supported combinations on Han, HALight, and Herc20.
+    pub async fn finalize_announced_swap<A, B>(
+        &mut self,
+        _announced: AnnouncedSwapParameters<A, B>,
+    ) -> anyhow::Result<AnnounceSwapParameters<A, B>> {
+        // let mut _swarm = self.swarm.lock().await;
+        // swarm.behaviour.finalize_announced_swap(_announced).await?
+
+        unimplemented!()
+    }
+}
+
+/// Holds the data that needs to be sent by Alice to Bob to announce a swap
+/// and take it from 'created' to 'finalized'.
+///
+/// For Han-Han, Han-Herc20, Herc20-Han
+/// - alpha_identity: refund identity for Alice
+/// - beta_identity: redeem identity for Alice
+///
+/// For Han-HALight
+/// - alpha_identity: refund identity for Alice (i.e., Ethereum address)
```

Comment:
Hum, what about cltv expiry? Do we know whether it's part of the `start` or `finalized` swap?

Reply:
I would assume it is part of the start because it is price-relevant, same as the expiries for existing protocols.

Reply:
To clarify @thomaseizinger comment - it is passed between the nodes _before_ the communication protocols start i.e., before we even announce a swap.